flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -2003,9 +2003,10 @@ struct CoordinateOpConversion
           // types and non contiguous arrays.
           auto idxTy = lowerTy().indexType();
           mlir::Value off = genConstantIndex(loc, idxTy, rewriter, 0);
+          unsigned dim = 0;
           for (unsigned index = i, lastIndex = i + arrTy.getDimension();
-               index < lastIndex; ++index) {
-            auto stride = loadStrideFromBox(loc, operands[0], index, rewriter);
+               index < lastIndex; ++index, ++dim) {
+            auto stride = loadStrideFromBox(loc, operands[0], dim, rewriter);
```

Comment:
I don't think you need to introduce a new variable here. Just replace the argument `index` with `index - i` should work, non?